arch/arm/core/thread.c

```diff
@@ -142,12 +142,21 @@ FUNC_NORETURN void _arch_user_mode_enter(k_thread_entry_t user_entry,
  *
  * This function configures per thread stack guards by reprogramming
  * the built-in Process Stack Pointer Limit Register (PSPLIM).
+ * The functionality is meant to be used during context switch.
  *
  * @param thread thread info data structure.
  */
 void configure_builtin_stack_guard(struct k_thread *thread)
 {
 #if defined(CONFIG_USERSPACE)
+	if (thread->arch.mode & CONTROL_nPRIV_Msk) {
+		/* Only configure stack limit for threads in privileged mode
+		 * (i.e supervisor threads or user threads doing system call).
```

Comment:
IMHO this explains it better:
... (i.e supervisor threads or user threads currently executing system call) ...